projects/kit/components/rating/rating.component.ts

```diff
@@ -0,0 +1,152 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    EventEmitter,
+    forwardRef,
+    HostListener,
+    Inject,
+    Input,
+    Output,
+} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+import {EMPTY_FUNCTION, moveFocus} from '@taiga-ui/cdk';
+
+import {RatingOptions, TUI_RATING_OPTIONS} from './rating.options';
+import {TuiRatingState} from './rating.state';
+import {getChildrenByParent} from './utils/get-children-by-parent';
+
+@Component({
+    selector: 'tui-rating',
+    templateUrl: './rating.template.html',
+    styleUrls: ['./rating.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => TuiRatingComponent),
+            multi: true,
+        },
+    ],
+})
+export class TuiRatingComponent implements ControlValueAccessor {
+    @Input() tuiRateMax: number = this.options.rateMax;
+    @Input() tuiRateReadonly: boolean = this.options.rateReadonly;
+    @Input() tuiRateColor: string = this.options.rateColor;
+    @Input() tuiRateEmptyIcon: string = this.options.rateEmptyIcon;
+    @Input() tuiRateSelectedIcon: string = this.options.rateSelectedIcon;
+    @Output() readonly tuiRateActiveChange: EventEmitter<number> = new EventEmitter();
+    ratingState: TuiRatingState = new TuiRatingState();
+    onTouched = EMPTY_FUNCTION;
+    onChange = EMPTY_FUNCTION;
+    touched = false;
+    disabled = false;
+
+    constructor(
+        @Inject(TUI_RATING_OPTIONS)
+        private readonly options: RatingOptions,
+        @Inject(ChangeDetectorRef) private readonly cd: ChangeDetectorRef,
+    ) {}
+
+    @Input()
+    set tuiRateDisable(isDisabled: boolean) {
+        this.disabled = isDisabled;
+    }
+
+    @Input()
+    set tuiRateActive(rate: number) {
+        this.ratingState.setCurrentRate(rate);
+    }
+
+    get enabled(): boolean {
+        return !(this.tuiRateReadonly || this.disabled);
+    }
+
+    @HostListener('keydown.arrowRight.prevent', ['$event.target', '1'])
+    @HostListener('keydown.arrowLeft.prevent', ['$event.target', '-1'])
+    onKeyDownArrow(current: HTMLElement, previousIndex: number): void {
```

Comment:
We don't need to navigate each item, let's just increase/decrease control value with arrows. If somebody doesn't want to make changes to the control on each step they can always set `updateOn: 'blur'`.

Reply:
But how then can we do so that we can navigate through each element without `onKeyDownArrow`?

https://mui.com/components/rating/

After all, popular components are supported keydown navigation

Reply:
You do not need to navigate, think about it as if it was `input type="range"`. Maybe it's even a good idea to put that input inside and make it transparent. This would probably make it more accessible 🤔